platform/engineering/design-docs/2020-12-14-cms-content-caching-strategy.md

```diff
@@ -0,0 +1,186 @@
+- [Update and publish the function code](https://docs.aws.amazon.com/lambda/latest/dg/API_UpdateFunctionCode.html) using the archived function from S3.
+- [Create the alias](https://docs.aws.amazon.com/lambda/latest/dg/API_CreateAlias.html) for the new version of the function.
+
+If the current branch is **not** `master`, [synchronously invoke](https://docs.aws.amazon.com/lambda/latest/dg/API_Invoke.html) the new version of the function (aliased with the checksum).
+
+If the current branch is `master`, also get the version that the `master` alias points to. If the master alias doesn't match the version of the checksum alias:
+- [Update the `master` alias](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/lambda/update-alias.html) to the version currently aliased with the checksum.
+- Synchronously invoke the new version of the function (aliased with `master`).
+
+Pull the cache from S3 that corresponds to the checksum (or `master` if on the `master` branch) for the build step.
+
+### Code Location
+
+Initially, the Lambda function and its supporting code will live in the `vets-website`.
+Once the content build is separated into the `content-build` repo, they will be moved there.
+
+Handler source code, bundling, and upload (`vets-website`):
+- Lambda function: `src/platform/lambdas/content-cache.js`
+- Webpack: `config/webpack.content-cache.js`
+- GitHub Actions: `.github/workflows/content-build.yml`
+- Function upload script: `scripts/upload-content-cache.sh`
+
+The Lambda function will be provisioned in the `devops` repo.
+- Terraform: `terraform/environments/global/lambda.tf`
+
+### Testing Plan
+
+The GitHub Actions pipeline will include a job for building and updating the function. This will alert us to any issues with building the function and provisioning it in AWS.
+
+The prototype for the function will simply download the data from the Drupal server. This is to verify that it can connect to the CMS at all.
+Once that is confirmed, the function will be further developed to cache the downloaded data to S3.
+
+When the function is fully written, we will monitor the log output in AWS CloudWatch to observe whether it works correctly.
+
+### Logging
+
+GitHub Actions will output logs for the build and versioning of the function.
+Log output from the function (from calling `console` methods) will be stored in CloudWatch.
+
+### Debugging
+
+One can start diagnosing any issues using the logs mentioned above.
+
+The Webpack build of the function can be run locally to debug issues with the build.
+
+To locally validate that the code properly pulls the Drupal data, the function can be run in a script with some modifications to disable the parts that are specific to the Lambda environment.
+
+Once the data is cached, it can be downloaded and used to build content. That build can be compared with a content build that pulls fresh data from Drupal with a SOCKS connection.
+
+### Caveats
+
+To be determined.
```

Comment:
@U-DON Can we elaborate on how the caveats will be determined and a rough timeline of when we'll start to figure that out?